usr/local/www/status_gateways.php

```diff
@@ -53,109 +53,116 @@
-				if ($gateways_status[$gname]) {
-					$status = $gateways_status[$gname];
-					if (stristr($status['status'], "force_down")) {
-						$online = gettext("Offline (forced)");
-						$bgcolor = "#F08080";  // lightcoral
-					} elseif (stristr($status['status'], "down")) {
-						$online = gettext("Offline");
-						$bgcolor = "#F08080";  // lightcoral
-					} elseif (stristr($status['status'], "loss")) {
-						$online = gettext("Warning, Packetloss").': '.$status['loss'];
-						$bgcolor = "#F0E68C";  // khaki
-					} elseif (stristr($status['status'], "delay")) {
-						$online = gettext("Warning, Latency").': '.$status['delay'];
-						$bgcolor = "#F0E68C";  // khaki
-					} elseif ($status['status'] == "none") {
-						$online = gettext("Online");
-						$bgcolor = "#90EE90";  // lightgreen
+			<?php $counter++; ?>
+			</td>
+			<td>
+<?php			if ($gateways_status[$gname])
+					echo $gateways_status[$gname]['loss'];
+				else
+					echo gettext("Pending");
+
+				$counter++;
+?>
+			</td>
+			<td>
+				<table>
+<?php
+					if ($gateways_status[$gname]) {
+						$status = $gateways_status[$gname];
+						if (stristr($status['status'], "force_down")) {
+							$online = gettext("Offline (forced)");
+							$bgcolor = "#F08080";  // lightcoral
+						} elseif (stristr($status['status'], "down")) {
+							$online = gettext("Offline");
+							$bgcolor = "#F08080";  // lightcoral
+						} elseif (stristr($status['status'], "loss")) {
+							$online = gettext("Warning, Packetloss").': '.$status['loss'];
+							$bgcolor = "#F0E68C";  // khaki
+						} elseif (stristr($status['status'], "delay")) {
+							$online = gettext("Warning, Latency").': '.$status['delay'];
+							$bgcolor = "#F0E68C";  // khaki
+						} elseif ($status['status'] == "none") {
+							$online = gettext("Online");
+							$bgcolor = "#90EE90";  // lightgreen
+						}
+					} else if (isset($gateway['monitor_disable'])) {
+							$online = gettext("Online");
+							$bgcolor = "#90EE90";  // lightgreen
+					} else {
+						$online = gettext("Pending");
+						$bgcolor = "#D3D3D3";  // lightgray
 					}
-				} else if (isset($gateway['monitor_disable'])) {
-						$online = gettext("Online");
-						$bgcolor = "#90EE90";  // lightgreen
-				} else {
-					$online = gettext("Pending");
-					$bgcolor = "#D3D3D3";  // lightgray
-				}
-				echo "<tr><td><table width='100%'><tr><td bgcolor=\"$bgcolor\">&nbsp;$online&nbsp;</td></tr><tr><td>";
-				$lastchange = $gateways_status[$gname]['lastcheck'];
-				if(!empty($lastchange)) {
-					echo gettext("Last check:") . '<br />' . $lastchange;
-				}
-				echo "</td></tr></table></td></tr>";
-                        ?>
-			</table>
-                  </td>
-		  <td class="listbg"> <?=$gateway['descr']; ?></td>
-                </tr>
-		<?php } ?>
-              </table>
-	   </div>
-</td></tr>
-</table>
+?>
+					<tr>
+						<td>
+							<table>
+								<tr>
+									<td bgcolor="<?=$bgcolor?>">&nbsp;<?=$online?>
```

Comment:
Thanks. I’ll look at that.

> On Apr 22, 2015, at 8:05 AM, SjonHortensius <notifications@github.com> wrote:
> 
> In usr/local/www/status_gateways.php <https://github.com/SjonHortensius/pfsense/pull/105#discussion_r28865706>:
> 
> > -                        ?>
> > -			</table>
> > -                  </td>
> > -		  <td class="listbg"> <?=$gateway['descr']; ?></td>
> > -                </tr>
> > -		<?php } ?>
> > -              </table>
> > -	   </div>
> > -</td></tr>
> > -</table>
> > +?>
> > +					<tr>
> > +						<td>
> > +							<table>
> > +								<tr>
> > +									<td bgcolor="<?=$bgcolor?>">&nbsp;<?=$online?>
> Why does this TD need to be nested in 3 tables? I propose you add the lastchange in the same cell. Also, it would be great if we could convert the background-color to an icon; did you have a look at the set for appropriate icons?
> 
> I'd rewrite this to simply <td>$online<i> last checked $lastcheck</i></td> instead of <td><table><tr><td><table><tr><td>
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/SjonHortensius/pfsense/pull/105/files#r28865706>.
> 

Steve
--
Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com <mailto:sbeaver@netgate.com> | www.netgate.com <http://www.netgate.com/>